Sources/GoogleAI/GenerativeModel.swift

```diff
@@ -96,11 +96,12 @@ public final class GenerativeModel {
   /// [few-shot](https://developers.google.com/machine-learning/glossary/generative#few-shot-prompting)
   /// prompts, see ``generateContent(_:)-58rm0``.
   ///
-  /// - Parameter content: The input(s) given to the model as a prompt (see ``PartsRepresentable``
+  /// - Parameter content: The input(s) given to the model as a prompt (see
+  /// ``ThrowingPartsRepresentable``
   /// for conforming types).
   /// - Returns: The content generated by the model.
   /// - Throws: A ``GenerateContentError`` if the request failed.
-  public func generateContent(_ parts: PartsRepresentable...)
+  public func generateContent(_ parts: any ThrowingPartsRepresentable...)
```

Comment:
Should `public func generateContent(_ content: [ModelContent]) async throws` (and `public func countTokens(_ content: [ModelContent]) async throws`) be updated in a similar to `public func generateContentStream(_ contentClosure: @autoclosure () throws -> [ModelContent])`?

I _think_ currently `promptImageContentError`s will only be thrown in `generateContentStream` but not `generateContent` or `countTokens` (but I could be wrong).

Reply:
Done, good catch.

Reply:
Thanks for adding the error handling to `generateContent`, Morgan! I think it's fine to leave it out of `countTokens` since we wouldn't want to throw `GenerateContentError.promptImageContentError(underlying: imageError)` there.